app/presenters/vendor_api/single_application_presenter.rb

```diff
@@ -255,6 +261,59 @@ def qualification_to_hash(qualification)
       }.merge HesaQualificationFieldsPresenter.new(qualification).to_hash
     end
 
+    def subject_code(qualification)
+      # There are a few cases here:
+      #
+      # Qualification level: gcse
+      #   These are either maths, science or english GCSEs that are required by some courses
+      #   - Maths GCSEs are not structured, so we map to a single maths subject code
+      #   - Science GCSEs are structured, but we combine them into one in the API response, so we can only be as granular as 'science'
+      #   - English GCSEs are now structured, but they did not used to be. We map to a generic english code here for the unstructured ones,
+      #     and have a better mapping within the structured_gcse_to_hashes method for newer english GCSEs
+      #
+      # Qualification level: other
+      #   These are other qualifications that a candidate has that are not direct requirements for the course.
+      #   This includes GCSEs and A levels, and none of them are structured. If their subject has been chosen
+      #   directly from the autocomplete, then we can map to a code. Otherwise, we leave the code blank.
```

Comment:
Is this comment useful here?

I often find qualifications really difficult to deal with because they are so inconsistent with how they hold data. We could add better documentation in a separate place, but I don't think it would be very visible.

Reply:
I would avoid large comments in the code. I think either have something in the commit message, as well as a smaller comment across the code relating to each subject, or maybe having a rethink of the code itself and if we can make it less complex/self documenting to help. But happy to have a chat around this specific case as Im not fully following the different cases